h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstFillNA.java

```diff
@@ -54,19 +54,25 @@ public Val apply(Env env, Env.StackHelp stk, AstRoot asts[]) {
     final String method = asts[2].exec(env).getStr();
     if (!(Arrays.asList("forward","backward")).contains(method.toLowerCase()))
       throw new IllegalArgumentException("Method must be forward or backward");
-    // Not impl yet
-    if (METHOD_BACKWARD.equalsIgnoreCase(method.trim()))
-      throw H2O.unimpl("Backward fillNA method is not yet implemented.");
 
     final int axis = (int) asts[3].exec(env).getNum();
     if (!(Arrays.asList(0,1)).contains(axis))
       throw new IllegalArgumentException("Axis must be 0 for columnar 1 for row");
     final int limit = (int) asts[4].exec(env).getNum();
+
+    if (limit == 0) // fast short cut to do nothing if user set zero limit
```

Comment:
The operation makes no sense if the limit is zero. This is for another Jira, but I'd validate the `limit` argument on the client already or at least on the backend. 

However, in current state, this is valuable here and I'd leave it. I'm just wondering of we should throw an exception and tell the user the operation has no sense at all ? Do we even validate it for limit < 0 ? An assertion for that could be considered here.

Reply:
I like your suggestion here.  We should throw an exception when limit < 0 and do nothing when limit == 0 at least for now.